libsolidity/formal/SymbolicState.cpp

```diff
@@ -149,6 +153,120 @@ smtutil::Expression SymbolicState::txConstraints(FunctionDefinition const& _func
 	return conj;
 }
 
+void SymbolicState::buildABIFunctions(set<FunctionCall const*> const& _abiFunctions)
+{
+	map<string, SortPointer> functions;
+
+	for (auto const* funCall: _abiFunctions)
+	{
+		auto t = dynamic_cast<FunctionType const*>(funCall->expression().annotation().type);
+
+		auto const& args = funCall->sortedArguments();
+		auto const& paramTypes = t->parameterTypes();
+		auto const& returnTypes = t->returnParameterTypes();
+
+
+		auto argTypes = [](auto const& _args) {
+			return applyMap(_args, [](auto arg) { return arg->annotation().type; });
+		};
+
```

Comment:
I guess I need to add some comments to this functions to help reading

Reply:
It for sure wouldn't hurt, but it's also not *too* bad :-).